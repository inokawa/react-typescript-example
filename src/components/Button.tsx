import React from 'react';

const component: React.FC<{
  title: string;
  onClick: () => void;
}> = ({ title, onClick }) => <button onClick={onClick}>{title}</button>;

export default component;
