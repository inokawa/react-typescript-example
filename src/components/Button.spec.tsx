import React from "react";
// @ts-expect-error FIXME
import renderer from "react-test-renderer";
import Button from "./Button";

it("sample", () => {
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  const component = renderer.create(<Button title="test" onClick={() => {}} />);
  expect(component.toJSON()).toMatchSnapshot();
});
