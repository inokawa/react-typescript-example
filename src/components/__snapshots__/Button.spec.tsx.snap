// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sample 1`] = `
<button
  onClick={[Function]}
>
  test
</button>
`;
